owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -445,13 +447,15 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
         val customTabsBuilder: CustomTabsIntent.Builder = CustomTabsIntent.Builder()
         val customTabsIntent: CustomTabsIntent = customTabsBuilder.build()
 
+        this.oidcState = OAuthUtils().generateRandomState()
         val authorizationEndpointUri = OAuthUtils.buildAuthorizationRequest(
             authorizationEndpoint = authorizationEndpoint,
             redirectUri = OAuthUtils.buildRedirectUri(applicationContext).toString(),
             clientId = clientId,
             responseType = ResponseType.CODE.string,
             scope = if (oidcSupported) OAUTH2_OIDC_SCOPE else "",
-            codeChallenge = oauthViewModel.codeChallenge
+            codeChallenge = oauthViewModel.codeChallenge,
+            state = this.oidcState
```

Comment:
I think that we could generate the state in the ViewModel as we do with `codeChallenge`. What do you think?

https://github.com/owncloud/android/blob/master/owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/oauth/OAuthViewModel.kt#L47

Reply:
Please have a look again, I just pushed the minor change, I hope this is how you meant it, at least it makes sense to me, looks more consistent now.